### PR TITLE
check for running DM for user switching

### PR DESCRIPTION
**Pull request by NiceandGently**
same as mate-session

### CODE REVIEW

**Comment by NiceandGently** [comment]
please cherry-pick to 1.8

File: src/gs-lock-plug.c [hunk 1]
```diff
@@ -248,7 +250,7 @@ do_user_switch (GSLockPlug *plug)
 			g_error_free (error);
 		}
 	}
-	else if (g_getenv ("XDG_SEAT_PATH") != NULL)
+	else if (process_is_running("lightdm"))
```
Comment (dnk) [inline src/gs-lock-plug.c]: check variable XDG_SEAT_PATH

File: src/gs-lock-plug.c [hunk 2]
```diff
@@ -1284,17 +1286,17 @@ gs_lock_plug_set_switch_enabled (GSLockPlug *plug,
...
 			gtk_widget_show (plug->priv->auth_switch_button);
 		}
-		else if (is_program_in_path (GDM_FLEXISERVER_COMMAND))
+		else if (process_is_running ("gdm") || process_is_running("gdm3"))
 		{
 			/* GDM */
 			gtk_widget_show (plug->priv->auth_switch_button);
 		}
-		else if (g_getenv ("XDG_SEAT_PATH") != NULL)
+		else if (process_is_running("lightdm"))
```
Comment (dnk) [inline src/gs-lock-plug.c]: and here

Comment (NiceandGently) [inline src/gs-lock-plug.c]: and what's the problem for checking if lightdm process is running?

**Pull request closed without merging**